metadetect/fitting.py

```diff
@@ -346,6 +346,92 @@ def _check_flags(self, mbobs):
         return isok
 
 
+class KSigmaMoments(Moments):
+    """
+    measure pre-PSF 'ksigma' weighted moments following Bernstein et al. but
+    in real-space.
+    """
+    def __init__(self, *args, **kw):
+        super(Moments, self).__init__(*args, **kw)
+        self._set_fitter()
+
+    def _print_result(self, data):
+        mess = "        ksigma s2n: %g Trat: %g"
+        logger.debug(mess % (data['ksigma_s2n'][0], data['ksigma_T_ratio'][0]))
+
+    def _measure_moments(self, obs):
+        """
+        measure weighted moments
+        """
+
+        if obs.has_psf():
+            res = self._fitter.go(obs)
+        else:
+            res = self._fitter.go(obs, no_psf=True)
+
+        res['numiter'] = 1
+        res['g'] = res['e']
+        res['g_cov'] = res['e_cov']
+        res['g_err'] = res['e_err']
```

Comment:
are you copying this out?  Should we add that to _measure_moments in the Moments class too?

Reply:
I removed this.